django/db/models/expressions.py

```diff
@@ -843,13 +843,19 @@ def as_sql(self, compiler, connection, template=None, case_joiner=None, **extra_
         template_params.update(extra_context)
         case_parts = []
         sql_params = []
+        from django.db.models.sql.datastructures import EmptyResultSet
```

Comment:
I think exceptions in `django.core.exceptions` are mostly things that a user might need to import. I don't think `EmptyResultSet` meets that criteria, at least in my experience I never needed it in a project.